src/shared/components/home/site-form.tsx

```diff
@@ -1141,6 +1195,36 @@ export class SiteForm extends Component<SiteFormProps, SiteFormState> {
     this.setState(s => ((s.siteForm.legal_information = Some(val)), s));
   }
 
+  handleTaglineChange(i: SiteForm, index: number, val: string) {
+    let taglines = i.state.siteForm.taglines.unwrap();
+    taglines[index] = val;
+    i.state.siteForm.taglines = Some(taglines);
```

Comment:
pre-emptively asking, is there a better way to modify an Option<string[]> by index beyond unwrapping and setting it directly like this?

Reply:
`.match` might work:

```ts
let taglines = i.state.siteForm.taglines.match({
  some: tls => { tls[index] = val; },
  none: void 0
});
i.state.siteForm.taglines = taglines;
```

If that doesn't work, at least do a `.isSome()` before doing the `unwrap()`... unwraps are always dangerous and can crash the UI on missing values.

Reply:
In this case, the handleChange/delete events will only be applicable on an already existing tagline, so it will always have some. Ill try out the match though and let you know how it goes.